### PR TITLE
Allow automatic credential fetch from access token

### PR DESCRIPTION
**Pull request by eliperelman**
This would allow you to do:

```js
import { Auth } from 'taskcluster-client-web';

const accessToken = 'get access token';
const auth = new Auth({ accessToken });

auth.currentScopes().then(({ scopes }) => /* do whatever */);
```

So now every client will await the fetching of credentials if an accessToken is provided.

### CODE REVIEW

**Comment by eliperelman** [comment]
Also note that you can override what URL you choose to fetch credentials from, as well as the underlying mechanism for fetching those credentials:

```js
new Client({
  exchangeAccessTokenUrl: String,

  exchangeAccessToken: Function<String accessToken, String exchangeAccessTokenUrl> ->
    Promise<Object<credentials, expires>>
});
```

**Comment by eliperelman** [comment]
@djmitche ready for review again.

File: README.md [hunk 1]
```diff
@@ -301,6 +301,19 @@ const auth = new Auth({
 });
 ```
 
+You may also choose to pass an access token from a Taskcluster-Login-supported authentication service,
+e.g. auth0. Passing an access token will cause the client instance to remotely fetch the associated
+credentials, which will then be filled in as options just as though they had been passed directly to
+the client constructor.
+
+Passing an access token to the client constructor also has the added benefit of automatically fetching
```
Comment (djmitche) [inline README.md]: s/also//

File: README.md [hunk 2]
```diff
@@ -301,6 +301,19 @@ const auth = new Auth({
...
+new credentials when the existing credentials expire, up until the access token's valid duration.
```
Comment (djmitche) [inline README.md]: This should probably have an example.  Let's leave the custom URL and exchange function undocumented for the moment, though

**Pull request merged**